### PR TITLE
drivers: wdt: espressif: Add 32K Xtal Watchdog

### PR DESCRIPTION
**Pull request by LucasTambor**
This WDT is responsible for monitoring the external 32.728 Hz crystal connected to pins XTAL_32K_P and XTAL_32K_N. 
If an oscillation failure is detected the hardware automatically switch to RTC_RC_SLOW clock source and triggers an interrupt.

### CODE REVIEW

File: drivers/clock_control/clock_control_esp32.c [hunk 1]
```diff
@@ -685,6 +687,15 @@ static int clock_control_esp32_init(const struct device *dev)
...
 
+#if defined(CONFIG_SOC_SERIES_ESP32S3)
+	uart_clock_src_hz = (uint32_t)rtc_clk_xtal_freq_get() * MHZ(1);
+#if !defined(ESP_CONSOLE_UART_NONE)
+	esp_rom_uart_set_clock_baudrate(ESP_CONSOLE_UART_NUM, uart_clock_src_hz,
+					ESP_CONSOLE_UART_BAUDRATE);
+#endif
+#endif
+
+
```
Comment (sylvioalves) [inline drivers/clock_control/clock_control_esp32.c:698]: please. remove extra line.

File: samples/boards/esp32/xt_wdt/README.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
+.. _xt_wdt_sample:
+
+Espressif XT WDT Sample
+#######################
+
+Overview
+********
+
+This sample shows how to work with XTAL32K Watchdog Timer (XT WDT) peripheral.
+To properly run this sample, you need to have a board with and external 32.728 Hz
```
Comment (sylvioalves) [inline samples/boards/esp32/xt_wdt/README.rst:10]: typo: `and` -> `an`

File: samples/boards/esp32/xt_wdt/README.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+crystal oscillator connected to the XTAL_32K_P and XTAL_32K_N pins.
+
+The app wil ask for the crystal removal to simulate a crystal failure and trigger
```
Comment (sylvioalves) [inline samples/boards/esp32/xt_wdt/README.rst:13]: typo: `wil`- > `will`.

File: samples/boards/esp32/xt_wdt/sample.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+sample:
+  description: Application to test XT WDT
+  name: xt_wdt_sample
+tests:
+  sample.board.esp32.xt_wdt:
+    platform_allow:
+      - esp32c3_devkitm
+      - esp32s2_saola
+      - esp32s3_devkitc_wrover/esp32s3/procpu
```
Comment (sylvioalves) [inline samples/boards/esp32/xt_wdt/sample.yaml:9]: esp32s3_devkitc_wrover/esp32s3/procpu should be esp32s3_devkitc/esp32s3/procpu

Comment (LucasTambor) [reply samples/boards/esp32/xt_wdt/sample.yaml]: I was aiming for esp32s3_devkitm/esp32s3_procpu haha

File: west.yml [hunk 5]
```diff
@@ -152,7 +152,7 @@ manifest:
       groups:
         - hal
     - name: hal_espressif
-      revision: b5f5fa6d08780c0077407150c6685d6c9d70262c
+      revision: dd06969eb722d041606b40d353bb23865b339044
```
Comment (sylvioalves) [inline west.yml:155]: Please. update revision here to 5191505f915c0b1c706222f4709925a453e0e858

File: drivers/clock_control/clock_control_esp32.c [hunk 6]
```diff
@@ -663,9 +662,12 @@ static int clock_control_esp32_configure(const struct device *dev, clock_control
 static int clock_control_esp32_init(const struct device *dev)
 {
 	const struct esp32_clock_config *cfg = dev->config;
+	struct esp32_clock_data *data = dev->data;
 	soc_reset_reason_t rst_reas;
 	rtc_config_t rtc_cfg = RTC_CONFIG_DEFAULT();
 	bool ret;
+	uint32_t uart_clock_src_hz;
+
```
Comment (marekmatej) [inline drivers/clock_control/clock_control_esp32.c:670]: Please remove the empty line.

**Review by sylvioalves — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]
LGTM

**Pull request merged**